internal/web/configurationvalues.go

```diff
@@ -72,10 +92,257 @@ func extractObjectKeyValueSource(r *http.Request, valueName string) *v1alpha1.Ob
+// In any case, the input container consists of a button where the user can change the type of reference or remove the
+// reference, and the actual input field(s).
+func (s *server) packageConfigurationInput(w http.ResponseWriter, r *http.Request) {
+	pkgName := mux.Vars(r)["pkgName"]
+	selectedVersion := r.FormValue("selectedVersion")
+	repositoryName := r.FormValue("repositoryName")
+	pkg, manifest, err := describe.DescribeInstalledPackage(r.Context(), pkgName)
+	if err != nil && !errors.IsNotFound(err) {
+		err = fmt.Errorf("an error occurred fetching package details of %v: %w", pkgName, err)
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		return
+	}
+
+	if manifest == nil {
+		manifest = &v1alpha1.PackageManifest{}
+		if err := s.repoClientset.ForRepoWithName(repositoryName).
+			FetchPackageManifest(pkgName, selectedVersion, manifest); err != nil {
+			// TODO check error handling again?
+			s.respondAlertAndLog(w, err,
+				fmt.Sprintf("An error occurred fetching manifest of %v in version %v", pkgName, selectedVersion),
+				"danger")
+			return
+		}
+	}
+
+	valueName := mux.Vars(r)["valueName"]
+	refKind := r.URL.Query().Get("refKind")
+	if valueDefinition, ok := manifest.ValueDefinitions[valueName]; ok {
+		options := pkg_config_input.PkgConfigInputDatalistOptions{}
+		if refKind == refKindConfigMap || refKind == refKindSecret {
+			if opts, err := s.getNamespaceOptions(); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to get namespace options: %v\n", err)
+			} else {
+				options.Namespaces = opts
+			}
+		} else {
+			if opts, err := s.getPackagesOptions(r.Context()); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to get package options: %v\n", err)
+			} else {
+				options.Names = opts
+			}
+		}
+		input := pkg_config_input.ForPkgConfigInput(
+			pkg, repositoryName, selectedVersion, pkgName, valueName, valueDefinition, nil, &options,
+			&pkg_config_input.PkgConfigInputRenderOptions{
+				Autofocus:      true,
+				DesiredRefKind: &refKind,
+			})
+		err = s.templates.pkgConfigInput.Execute(w, input)
+		checkTmplError(err, fmt.Sprintf("package config input (%s, %s)", pkgName, valueName))
+	}
+}
+
+// namesDatalist is a GET endpoint returning an html datalist, containing options depending on the given valueName,
+// kind of reference and namespace. It is only usable for ConfigMap and Secret refs, since packages don't have a
+// namespace. In case the refKind is ConfigMap, the datalist contains the config maps of the given namespace; in case
+// the refKind is Secret, the datalist contains the secrets of the given namespace; if no namespace is given or an
+// error occurs, an empty datalist is returned
+func (s *server) namesDatalist(w http.ResponseWriter, r *http.Request) {
+	valueName := mux.Vars(r)["valueName"]
+	refKind := r.FormValue(refKindKey)
+	id := r.FormValue("id")
+	nsKey := formKey(valueName, namespaceKey)
+	namespace := r.Form.Get(nsKey)
+	var options []string
+	if err := r.ParseForm(); err != nil {
```

Comment:
I don't understand why this is necessary here. Shouldn't the form be parsed before it is accessed using `FormValue`? Or am I missing something?